runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TranslationUtils.java

```diff
@@ -159,6 +159,18 @@ public T2 call(Tuple2<T1, T2> v1) throws Exception {
         };
       }
 
+  /** Extract window from a {@link KV} with {@link WindowedValue} value. */
+  static <K, V> Function<KV<K, WindowedValue<V>>, WindowedValue<KV<K, V>>> toKVByWindowInValue() {
+    return new Function<KV<K, WindowedValue<V>>, WindowedValue<KV<K, V>>>() {
+      @Override public WindowedValue<KV<K, V>> call(KV<K, WindowedValue<V>> kwv) throws Exception {
+        K k = kwv.getKey();
+        WindowedValue<V> wv = kwv.getValue();
+        V v = wv.getValue();
+        return WindowedValue.of(KV.of(k, v), wv.getTimestamp(), wv.getWindows(), wv.getPane());
```

Comment:
Try: ` wv.withValue(KV.of(kv.getKey(), wv.getValue()))` instead.
This is used in a couple of other places such as [combinePerKey](https://github.com/apache/beam/blob/master/runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TransformTranslator.java#L327) so you could look them up and use this utility there as well.